### PR TITLE
update parser tests to be generic

### PR DESCRIPTION
**Pull request by bakerboy448**
#### Database Migration
NO

#### Description
Title

#### Todos
- [X] Tests
- Wiki Updates-N/A

#### Issues Fixed or Closed by this PR

*

### CODE REVIEW

File: src/NzbDrone.Core.Test/ParserTests/UnicodeReleaseParserFixture.cs [hunk 1]
```diff
@@ -80,7 +80,7 @@ public void should_parse_unicode_digits(string postTitle, string title, int abso
             result.FullSeason.Should().BeFalse();
         }
 
-        [TestCase("[BeanSub][Nanatsu_no_Taizai_Fundo_no_Shinpan][01][GB][1080P][x264_AAC]", "Nanatsu no Taizai Fundo no Shinpan", "BeanSub", 1)]
+        [TestCase("[BeanSub][Anime Series Title][01][GB][1080P][x264_AAC]", "Anime Series Title", "BeanSub", 1)]
```
Comment (Taloth) [inline src/NzbDrone.Core.Test/ParserTests/UnicodeReleaseParserFixture.cs:83]: Keep underscores.

File: src/NzbDrone.Core.Test/ParserTests/UnicodeReleaseParserFixture.cs [hunk 2]
```diff
@@ -53,9 +53,9 @@ public void should_parse_unbracketed_chinese_anime_releases(string postTitle, st
             result.FullSeason.Should().BeFalse();
         }
 
-        [TestCase("[YMDR][慕留人 -火影忍者新時代-][Boruto -Naruto Next Generations-][2017][88-91][1080p][AVC][JAP][BIG5][MP4-AAC][繁中]", "Boruto -Naruto Next Generations", "YMDR", new[] { 88, 89, 90, 91 })]
-        [TestCase("[诸神字幕组][战栗杀机][BANANA FISH][01-24完][简日双语字幕][720P][MP4]", "BANANA FISH", "诸神字幕组", new[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24 })]
-        //[TestCase("【漫貓&愛戀字幕組】[五等分的新娘/五等分的花嫁/五等分の花嫁][Go-Toubun_no_Hanayome][01_03][BIG5][720P][HEVC]", "Go-Toubun no Hanayome", "漫貓&愛戀字幕組", new[] { 1, 2, 3 })]
+        [TestCase("[YMDR][慕留人 -火影忍者新時代-][Anime Series Title-][2017][88-91][1080p][AVC][JAP][BIG5][MP4-AAC][繁中]", "Anime Series Title", "YMDR", new[] { 88, 89, 90, 91 })]
+        [TestCase("[诸神字幕组][战栗杀机][Anime Series Title][01-24完][简日双语字幕][720P][MP4]", "Anime Series Title", "诸神字幕组", new[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24 })]
```
Comment (Taloth) [inline src/NzbDrone.Core.Test/ParserTests/UnicodeReleaseParserFixture.cs:57]: Keep upper case

File: src/NzbDrone.Core.Test/ParserTests/UnicodeReleaseParserFixture.cs [hunk 3]
```diff
@@ -37,10 +37,10 @@ public void should_parse_chinese_anime_releases(string postTitle, string title,
...
         }
 
-        [TestCase("[喵萌奶茶屋&LoliHouse]玛娜利亚魔法学院/巴哈姆特之怒Manaria Friends - 03 [WebRip 1080p HEVC-10bit AAC][简繁内封字幕]", "巴哈姆特之怒Manaria Friends", "喵萌奶茶屋&LoliHouse", 3)]
-        [TestCase("[悠哈璃羽字幕社&拉斯观测组&LoliHouse] 刀剑神域: Alicization / Sword Art Online: Alicization - 17 [WebRip 1080p HEVC-10bit AAC][简繁内封字幕]", "Sword Art Online Alicization", "悠哈璃羽字幕社&拉斯观测组&LoliHouse", 17)]
-        [TestCase("[ZERO字幕組]嫁給非人類·Jingai-san no Yome[11][BIG5][1080p]", "Jingai-san no Yome", "ZERO字幕組", 11)]
-        [TestCase("[Lilith-Raws] 艾梅洛閣下 II 世事件簿 -魔眼蒐集列車 Grace note- / Lord El-Melloi II Case Files - 04 [BiliBili][WEB-DL][1080p][AVC AAC][CHT][MKV]", "Lord El-Melloi II Case Files", "Lilith-Raws", 4)]
+        [TestCase("[喵萌奶茶屋&LoliHouse]玛娜利亚魔法学院/巴哈姆特之怒Anime Series Title - 03 [WebRip 1080p HEVC-10bit AAC][简繁内封字幕]", "巴哈姆特之怒Anime Series Title", "喵萌奶茶屋&LoliHouse", 3)]
+        [TestCase("[悠哈璃羽字幕社&拉斯观测组&LoliHouse] 刀剑神域: Alicization / Anime Series: Title - 17 [WebRip 1080p HEVC-10bit AAC][简繁内封字幕]", "Anime Series Title", "悠哈璃羽字幕社&拉斯观测组&LoliHouse", 17)]
+        [TestCase("[ZERO字幕組]嫁給非人類·Anime Series Title[11][BIG5][1080p]", "Anime Series Title", "ZERO字幕組", 11)]
```
Comment (Taloth) [inline src/NzbDrone.Core.Test/ParserTests/UnicodeReleaseParserFixture.cs:42]: Keep dash.

File: src/NzbDrone.Core.Test/ParserTests/DailyEpisodeParserFixture.cs [hunk 4]
```diff
@@ -11,27 +11,27 @@ namespace NzbDrone.Core.Test.ParserTests
...
+        [TestCase("2020.A.Late.Talk.Show.2012.16.02.PDTV.XviD-C4TV", "2020 A Late Talk Show", 2012, 2, 16)]
+        [TestCase("2020.A.Late.Talk.Show.2012.13.02.PDTV.XviD-C4TV", "2020 A Late Talk Show", 2012, 2, 13)]
+        [TestCase("2020.A.Late.Talk.Show.2011.12.02.PDTV.XviD-C4TV", "2020 A Late Talk Show", 2011, 12, 2)]
         [TestCase("Series Title - 2013-10-30 - Episode Title (1) [HDTV-720p]", "Series Title", 2013, 10, 30)]
-        [TestCase("The_Voice_US_04.28.2014_hdtv.x264.Poke.mp4", "The Voice US", 2014, 4, 28)]
-        [TestCase("At.Midnight.140722.720p.HDTV.x264-YesTV", "At Midnight", 2014, 07, 22)]
-        [TestCase("At_Midnight_140722_720p_HDTV_x264-YesTV", "At Midnight", 2014, 07, 22)]
+        [TestCase("The_Series_US_04.28.2014_hdtv.x264.Poke.mp4", "The Series US", 2014, 4, 28)]
+        [TestCase("A.Late.Talk.Show.140722.720p.HDTV.x264-YesTV", "A Late Talk Show", 2014, 07, 22)]
+        [TestCase("A.Late.Talk.Show_140722_720p_HDTV_x264-YesTV", "A Late Talk Show", 2014, 07, 22)]
```
Comment (Taloth) [inline src/NzbDrone.Core.Test/ParserTests/DailyEpisodeParserFixture.cs:27]: Keep underscore.

File: src/NzbDrone.Core.Test/ParserTests/AnimeMetadataParserFixture.cs [hunk 5]
```diff
@@ -8,21 +8,21 @@ namespace NzbDrone.Core.Test.ParserTests
...
-        [TestCase("[K-F] One Piece 214 10x14", "K-F", "")]
-        [TestCase("Bleach - 031 - The Resolution to Kill [Lunar].avi", "Lunar", "")]
-        [TestCase("[ACX]Hack Sign 01 Role Play [Kosaka] [9C57891E].mkv", "ACX", "9C57891E")]
-        [TestCase("[S-T-D] Soul Eater Not! - 06 (1280x720 10bit AAC) [59B3F2EA].mkv", "S-T-D", "59B3F2EA")]
+        [TestCase("[SubDESU]_Show_Title_DxD_07_(1280x720_x264-AAC)_[6B7FD717]", "SubDESU", "6B7FD717")]
+        [TestCase("[Chihiro]_Show_Title!!_-_06_[848x480_H.264_AAC][859EEAFA]", "Chihiro", "859EEAFA")]
+        [TestCase("[Underwater]_Show_Title_-_12_(720p)_[5C7BC4F9]", "Underwater", "5C7BC4F9")]
+        [TestCase("[HorribleSubs]_Show_Title_-_33_[720p]", "HorribleSubs", "")]
+        [TestCase("[HorribleSubs] Show-Title - 13 [1080p].mkv", "HorribleSubs", "")]
+        [TestCase("[Doremi].Show.Title.31.[1280x720].[C65D4B1F].mkv", "Doremi", "C65D4B1F")]
```
Comment (Taloth) [inline src/NzbDrone.Core.Test/ParserTests/AnimeMetadataParserFixture.cs:16]: Keep "5 Go Go!" It's what can throw off the parser. Same for the ones below

File: src/NzbDrone.Core.Test/ParserTests/AnimeMetadataParserFixture.cs [hunk 6]
```diff
@@ -8,21 +8,21 @@ namespace NzbDrone.Core.Test.ParserTests
...
+        [TestCase("[Doremi].Show.Title.31.[1280x720].[C65D4B1F]", "Doremi", "C65D4B1F")]
+        [TestCase("[Doremi].Show.Title.31.[1280x720].mkv", "Doremi", "")]
+        [TestCase("[K-F] Series Title 214", "K-F", "")]
+        [TestCase("[K-F] Series Title S10E14 214", "K-F", "")]
+        [TestCase("[K-F] Series Title 10x14 214", "K-F", "")]
+        [TestCase("[K-F] Series Title 214 10x14", "K-F", "")]
+        [TestCase("Series Title - 031 - The Resolution to Kill [Lunar].avi", "Lunar", "")]
+        [TestCase("[ACX]Series Title 01 Episode Name [Kosaka] [9C57891E].mkv", "ACX", "9C57891E")]
+        [TestCase("[S-T-D] Series Title - 06 (1280x720 10bit AAC) [59B3F2EA].mkv", "S-T-D", "59B3F2EA")]
```
Comment (Taloth) [inline src/NzbDrone.Core.Test/ParserTests/AnimeMetadataParserFixture.cs:25]: Keep exclamation mark.

**Review by Taloth — CHANGES_REQUESTED** [review]
Some comments, I probably didn't commented on all discrepancies. I like to stick as much as possible to the original punctuation, so the testcases remain diverse.

File: src/NzbDrone.Core.Test/ParserTests/UnicodeReleaseParserFixture.cs [hunk 7]
```diff
@@ -13,18 +13,18 @@ namespace NzbDrone.Core.Test.ParserTests
...
         [TestCase("[愛戀&漫貓字幕组][10月新番][關於我轉生後成爲史萊姆那件事][18][720P][BIG5][MP4]", "關於我轉生後成爲史萊姆那件事", "愛戀&漫貓字幕组", 18)]
-        [TestCase("【咕咕茶字幕組】★1月新番[天使降臨到了我身邊! / Watashi ni Tenshi ga Maiorita!][04][1080P][繁體][MP4]", "Watashi ni Tenshi ga Maiorita!", "咕咕茶字幕組", 4)]
-        [TestCase("【千夏字幕组】【天使降临到了我身边！_Watashi ni Tenshi ga Maiorita!】[第05话][1080p_HEVC][简繁外挂]", "Watashi ni Tenshi ga Maiorita!", "千夏字幕组", 5)]
-        [TestCase("【DHR动研字幕组】[多田君不恋爱_Tada-kun wa Koi wo Shinai][13完][繁体][720P][MP4]", "Tada-kun wa Koi wo Shinai", "DHR动研字幕组", 13)]
-        [TestCase("【动漫国字幕组】★01月新番[Endro～！][01][1080P][简体][MP4]", "Endro～！", "动漫国字幕组", 1)]
+        [TestCase("【咕咕茶字幕組】★1月新番[天使降臨到了我身邊! / Anime Series Title!][04][1080P][繁體][MP4]", "Anime Series Title!", "咕咕茶字幕組", 4)]
+        [TestCase("【千夏字幕组】【天使降临到了我身边！_Anime Series Title】[第05话][1080p_HEVC][简繁外挂]", "Anime Series Title", "千夏字幕组", 5)]
+        [TestCase("【DHR动研字幕组】[多田君不恋爱_Anime Series Title][13完][繁体][720P][MP4]", "Anime Series Title", "DHR动研字幕组", 13)]
+        [TestCase("【动漫国字幕组】★01月新番[Anime Series Title~!][01][1080P][简体][MP4]", "Anime Series Title~!", "动漫国字幕组", 1)]
```
Comment (Taloth) [inline src/NzbDrone.Core.Test/ParserTests/UnicodeReleaseParserFixture.cs:27]: Is this '!' or '！'? (yes, both are 1 character.

Comment (bakerboy448) [reply src/NzbDrone.Core.Test/ParserTests/UnicodeReleaseParserFixture.cs:27]: `~!` => `～！`

**Pull request merged**